scripts/js/lib/links/ExternalLink.ts

```diff
@@ -58,16 +58,22 @@ export class ExternalLink {
   }
 }
 
-function getHeaders(link: string) {
+export function getHeaders(link: string): HeadersInit {
   const headers: HeadersInit = {
-    "User-Agent": "qiskit-documentation-broken-links-finder",
+    "User-Agent": "qiskit-documentation-scripts",
   };
 
-  if (link.startsWith("https://github.com")) {
+  if (
+    link.startsWith("https://github.com") ||
+    link.startsWith("https://api.github.com")
+  ) {
     const ghToken = process.env.GITHUB_TOKEN;
-    if (ghToken) {
-      headers["Authorization"] = `token ${ghToken}`;
+    if (!ghToken) {
+      throw new Error(
+        "The env variable GITHUB_TOKEN is not set. You can get your token by running `gh auth token`.",
+      );
     }
```

Comment:
If the token isn't set, you'll get a 403 when generating the dev docs from a GitHub URL. This is relevant for when you do e.g. `regen-apis` locally. So, we should error eagerly.